spec/apis/event-listeners.spec.js

```diff
@@ -0,0 +1,69 @@
+export function notStartedEventListeners() {
+	describe(`event listeners before single-spa is started :`, () => {
+		beforeEach(ensureCleanSlate)
+		
+		it(`calls hashchange and popstate event listeners even when single-spa is not started`, done => {
+			let hashchangeCalled = false, popstateCalled = false;
+			window.addEventListener("hashchange", () => {
+				if (window.location.hash === '#/a-new-hash')
+					hashchangeCalled = true;
+			});
+			window.addEventListener("popstate", () => {
+				if (window.location.hash === '#/a-new-hash')
+					popstateCalled = true;
+			});
+
+			window.location.hash = '#/a-new-hash';
+			setTimeout(() => {
+				expect(hashchangeCalled).toBe(true);
+				expect(popstateCalled).toBe(true);
+				done();
+			});
+		});
+	});
+}
+
+export function yesStartedEventListeners() {
+	describe(`event listeners after single-spa is started`, () => {
+		beforeEach(ensureCleanSlate);
+
+		it(`calls all of the enqueued hashchange listeners even when the first event given to singleSpa is a popstate event`, done => {
```

Comment:
This test fails without the changes to reroute.js. It ensures that all enqueued event listeners are called.